src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -179,6 +179,19 @@ private static void SplitIRAndState(ControlInfoJson.Item control, string topPare
                 Functions = functions,
             };
 
+            // Storing Template PCFDynamicSchemaForIRRetrieval/OverridableProperties field values for each control instance.
+            // Since this value could be different for each control instance though it follows same control template.
+            // Eg:Control Instance 1 -> template1 -> PCFDynamicSchemaForIRRetrieval1/OverridableProperties1
+            // Control Instance 2 -> template1 -> PCFDynamicSchemaForIRRetrieval2/OverridableProperties2
+
+            if (control.Template.ExtensionData.TryGetValue("PCFDynamicSchemaForIRRetrieval", out object val))
+            {
+                entropy.PCFDynamicSchemaForIRRetrievalEntry.Add(control.Name, val);
+            }
+            if (control.Template.ExtensionData.TryGetValue("OverridableProperties", out object val1))
```

Comment:
nit: Do we want to be more specific with variables names instead of adding numbers to them? Maybe changing the name of 'val' and 'val1' to something like 'PDFVal' and 'propertyVal'?

Reply:
Good catch! makes sense, need to clean up the names. Forgot to change.